### PR TITLE
Rewrite kyber extraction script in python

### PR DESCRIPTION
**Pull request by xvzcf**

### CODE REVIEW

File: .github/workflows/hax.yml [hunk 1]
```diff
@@ -50,4 +50,5 @@ jobs:
       - name: 🏃🏻‍♀️ Run hax extraction
         run: |
           eval $(opam env)
-          ./proof.sh extract
+          python3 extract_to_fstar.py --kyber-reference
+          python3 extract_to_fstar.py --crate specs/kyber
```
Comment (franziskuskiefer) [inline .github/workflows/hax.yml:54]: I would expect `./extract_to_fstar.py --crate hacspec_kyber`. `spec/kyber` is not a crate but the path to a crate.

Comment (franziskuskiefer) [inline .github/workflows/hax.yml:54]: Can you add something here as well to extract with the `--functions` argument? To make sure that's working.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks, certainly better than bash ...

Running `./extract_to_fstar.py` doesn't appear to do anything? Or is it trying to extract everything and is just slow? It should probably show a help message at the moment.

Anyway, we need to either get the cli output of the extraction command or some progress bar or something to know what's happening.

**Comment by xvzcf** [comment]
> Running `./extract_to_fstar.py` doesn't appear to do anything? Or is it trying to extract everything and is just slow? It should probably show a help message at the moment.

You should get the following output when running just that:

```
Command: cargo hax into fstar 
Directory: .
```

File: .github/workflows/hax.yml [hunk 2]
```diff
@@ -50,4 +50,5 @@ jobs:
       - name: 🏃🏻‍♀️ Run hax extraction
         run: |
           eval $(opam env)
-          ./proof.sh extract
+          python3 extract_to_fstar.py --kyber-reference
```
[the remark this answers is not in the document]
Comment (xvzcf) [reply .github/workflows/hax.yml:53]: I'd rather leave it as is, we can also extract the Kyber specification but that is through specifying the normal arguments. Only this requires special handling.

File: extract_to_fstar.py
[the remark this answers is not in the document]
Comment (xvzcf) [reply extract_to_fstar.py:1]: Done.

File: .github/workflows/hax.yml (hunk 1 above)
Comment (xvzcf) [reply .github/workflows/hax.yml:54]: Renamed the argument to `crate-path`.

File: extract_to_fstar.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    nargs="?",
+    default=".",
+    help="Path to the crate from which to extract the code (default is path to libcrux).",
+)
+parser.add_argument(
+    "--functions",
+    type=str,
+    nargs="*",
+    default="",
+    help="Space-separated list of functions to extract",
```
[the remark this answers is not in the document]
Comment (xvzcf) [reply extract_to_fstar.py:45]: Added a `modules` argument.

**Review by franziskuskiefer — APPROVED** [review]
Thanks, lets get it in.

**Pull request merged**